packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics-batcher.ts

```diff
@@ -32,174 +22,173 @@ describe('plugin-metrics', () => {
         },
       });
 
-      webex.config.metrics = config.metrics;
-
-      webex.request = function (options) {
-        return Promise.resolve({
-          statusCode: 204,
-          body: undefined,
-          options,
-        });
-      };
+      webex.request = (options) => Promise.resolve({body: {items: []}, options});
       sinon.spy(webex, 'request');
-    });
 
-    let clock;
+      NewMetrics.initialSetupCallDiagnosticMetrics({}, webex);
 
-    beforeEach(() => {
-      //@ts-ignore
-      clock = FakeTimers.install();
+      webex.config.metrics = config.metrics;
     });
 
     afterEach(() => {
-      clock.uninstall();
+      sinon.restore();
     });
 
     describe('#request()', () => {
-      describe('when the request completes successfully', () => {
-        it('clears the queue', () => {
-          clock.uninstall();
-
-          return webex.internal.metrics.callDiagnosticEventsBatcher
-            .request({
-              type: 'diagnostic-event',
-              eventPayload: {
-                originTime: {
-                  triggered: 'mock triggered timestamp',
-                },
-              },
-            })
-            .then(() => {
-              //@ts-ignore
-              assert.calledOnce(webex.request);
-              assert.lengthOf(webex.internal.metrics.callDiagnosticEventsBatcher.queue, 0);
-            });
+      describe('when the request completes successfully', async () => {
+        it('clears the queue', async () => {
+          await NewMetrics.callDiagnosticMetrics.submitToCallDiagnostics(
+            //@ts-ignore
+            {event: {name: 'client.interstitial-window.launched'}}
+          );
+          await flushPromises();
+
+          //@ts-ignore
+          assert.calledOnce(webex.request);
+          assert.lengthOf(webex.internal.metrics.callDiagnosticEventsBatcher.queue, 0);
         });
-      });
 
-      describe('when the request fails due to network disconnect', () => {
```

Comment:
this test has been removed and I don't see a new equivalent tests for it, do we not need it anymore?

Reply:
I honestly don't see any logic for this in the batcher file, i think it's testing functionality from the inherited Batcher class, and that's why I removed it. (I think it doesn't belong here).